### PR TITLE
IVY-1483 Invoke Ant through Ivy;

### PR DESCRIPTION
**Pull request by twogee**
like this: `java -jar ivy.jar -properties version.properties -main org.apache.tools.ant.Main`

### CODE REVIEW

File: ivy.xml [hunk 1]
```diff
@@ -61,6 +61,7 @@
         <dependency org="org.hamcrest" name="hamcrest-library" rev="${hamcrest.version}" conf="test"/>
         <dependency org="org.apache.ant" name="ant-testutil" rev="${apache-ant.version}" conf="test" transitive="false"/>
         <dependency org="org.apache.ant" name="ant-launcher" rev="${apache-ant.version}" conf="test" transitive="false"/>
+        <dependency org="org.apache.ant" name="ant-junit" rev="${apache-ant.version}" conf="test" transitive="false"/>
```
Comment (bodewig) [inline ivy.xml]: why has this dependency been added? I don't see it used.

Comment (twogee) [reply ivy.xml]: If Ant (launcher and core) is fetched as a dependency, then ant-junit must be retrieved as a dependency, too, in order to run unit tests.

Comment (bodewig) [reply ivy.xml]: to run which unit tests? Ivy's?

Comment (twogee) [reply ivy.xml]: Yes. Invocation of Ant runs Ivy's build.xml; it's a fancy way of bootstrapping an Ivy build, by the looks of it.

Comment (bodewig) [reply ivy.xml]: strange, thanks. As Ivy uses junit4 you will also need ant-junit4 in this case.

Comment (twogee) [reply ivy.xml]: Good catch, thanks. BTW, run this way Ant core seems unable to simply find org.apache.tools.ant.taskdefs.optional.junit.JUnitTask on a classpath; why is that? does it need some hint from the launcher?

Comment (twogee) [reply ivy.xml]: Ah, I get it... the way things work, ant-junit4 must declare a dependency on ant-junit

**Pull request closed without merging**